evalml/automl/automl_search.py

```diff
@@ -489,7 +491,10 @@ def search(self, X, y, data_checks="auto", feature_types=None, show_iteration_pl
                     desc = desc[:self._MAX_NAME_LEN - 3] + "..."
                 desc = desc.ljust(self._MAX_NAME_LEN)
 
-                update_pipeline(logger, desc, len(self._results['pipeline_results']) + 1, self.max_iterations, self._start)
+                if self.max_batches:
+                    update_pipeline(logger, desc, len(self._results['pipeline_results']) + 1, self.max_iterations, self._start, self._automl_algorithm.batch_number)
```

Comment:
@christopherbunn I see we're passing this batch number directly to the output. Does it start at 0, or at 1? RE someone else's comment, I think the batch should start at 1 in the log output.

Reply:
Aha, I read further. Here's what I suggest:
* Everywhere we call `update_pipeline`, pass the batch number in using `self._automl_algorithm.batch_number`. This will be 0 for the first batch
* In `update_pipeline`, update the format string to print `{current_batch+1}` instead of `{current_batch}`

Sound good?

We should also add unit test coverage which checks that the right batch number prints out. My suggestion:
* Set `max_iterations = len(get_estimators(self.problem_type))+1 + _pipelines_per_batch*2` to run 3 batches
* Mock pipeline fit/score to avoid long runtime
* Run search, and using the `caplog` fixture, assert that `Batch 1` appears `len(get_estimators(self.problem_type)) + 1` times, and `Batch 2` and `Batch 3` each appear `_pipelines_per_batch` times.

Reply:
The test case sounds good, I'll add that in. 

> In update_pipeline, update the format string to print {current_batch+1} instead of {current_batch}

@dsherry because the pipeline is counted as its own separate batch, the number of batches would actually be off by one since `batch_number` is incremented by 1 when `next_batch()` is called. Since `next_batch()` is called for the first batch before the output is seen, Batch 1 actually shows up as Batch 2 with `{current_batch+1}`.

I could:
1.  Keep `{current_batch+1}` and move where we increment batch number to when the results are reported back
2. Have the baseline pipeline show as batch 0 and use `self._automl_algorithm.batch_number`  
3. Override the output for the baseline (and not use `self._automl_algorithm.batch_number`) and show it as either batch 1, "baseline",  or "base"

Imo 1 has the potential to get messy since we're modifying `IterativeAlgorithm` so I want to avoid that. I'm leaning towards 2 although 3 seems fine with me as well. If we wanted "base" or "baseline" for 3, then I'll add another case for `update_pipeline()`

Reply:
Ah got it. Thanks.

I'll throw one more option on the pile:
2b. Label the baseline pipeline as batch 1 instead of batch 0. Use `self._automl_algorithm.batch_number` everywhere else, meaning the 1st batch shows up as batch 1, etc.

I think I agree with what you had here, and we should keep the baseline pipeline displayed as "batch 1" along with the others.

Sound good?

Reply:
2b works for me, I'll manually set the baseline as batch 1 then. Thanks!